opm/simulators/flow/EclWriter.hpp

```diff
@@ -350,19 +350,24 @@ class EclWriter : public EclGenericWriter<GetPropType<TypeTag, Properties::Grid>
     //! \brief Writes the initial FIP report as configured in RPTSOL.
     void writeInitialFIPReport()
     {
-        const auto& fip = simulator_.vanguard().eclState().getEclipseConfig().fip();
-        if (!fip.output(FIPConfig::OutputField::FIELD) &&
-            !fip.output(FIPConfig::OutputField::RESV)) {
-            return;
-        }
-
         const auto& gridView = simulator_.vanguard().gridView();
         const int num_interior = detail::
             countLocalInteriorCellsGridView(gridView);
 
         this->outputModule_->
             allocBuffers(num_interior, 0, false, false, /*isRestart*/ false);
 
+        // We always calculate the initial fip values as it may be used by various
+        // keywords in the Schedule, e.g. FIP=2 in RPTSCHED but no FIP in RPTSOL
+        outputModule_->calc_initial_inplace(simulator_.gridView().comm());
```

Comment:
Oh, it just occurred to me that this call might be happening a little too early.  If we don't populate the buffers allocated in `allocBuffers()`&ndash;by calling [`updateFluidInPlace()`](https://github.com/OPM/opm-simulators/blob/dea336a25daa52c7e6cee2920c53e2abac51aae1/opm/simulators/flow/EclWriter.hpp#L378)&ndash;then we risk that the initial fluid volumes are all zero.  Please verify that that does not happen now.

Reply:
nice catch. indeed I need to move that block as well. and then I realized I can simplify this whole thing, see new commit.

Reply:
> indeed I need to move that block as well

Cool&ndash;I'm happy we caught that.

>. and then I realized I can simplify this whole thing, see new commit.

Nice!